### PR TITLE
format: enforce pragma once for header files

### PR DESCRIPTION
**Pull request by botengyao**
format: enforce pragma once for header files and add auto fix support.

https://github.com/envoyproxy/envoy/issues/15264

Commit Message:
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue]
[Optional Fixes commit #PR or SHA]
[Optional Deprecated:]
[Optional [API Considerations](https://github.com/envoyproxy/envoy/blob/main/api/review_checklist.md):]

### CODE REVIEW

**Comment by botengyao** [comment]
There maybe some files that doesn't like `pragma once`, pending check.

File: tools/code_format/check_format.py [hunk 1]
```diff
@@ -159,6 +159,16 @@ def evaluate_lines(self, path, line_xform, write=True):
         format_flag = True
         output_lines = []
         for line_number, line in enumerate(self.read_lines(path)):
+            if line_number == 0 and path.endswith(".h"):
+                if line == "":
+                    error_message = "%s:%d: %s" % (
+                        path, line_number + 1, "the first line is empty for header files")
+                    continue
```
Comment (ravenblackx) [inline tools/code_format/check_format.py:166]: Isn't this part redundant (and a weird unhelpful error message that will likely lead to getting the one about `#pragma` afterwards)?

Why not drop this condition and just have the error message on the other one be `"#pragma once must be the first line of header files"`

Comment (ravenblackx) [reply tools/code_format/check_format.py:166]: (Realizing it's maybe to do with the autocorrection.)

Comment (phlax) [inline tools/code_format/check_format.py:166]:
```suggestion
            if line_number > 0 or not path.endswith(".h"):
                break
            if line == "":
                error_message = "%s:%d: %s" % (
                    path, line_number + 1, "the first line is empty for header files")
```

assuming the `line_numbers` are coming in order, we want to stop iterating as soon as they are `> 0`

Comment (phlax) [reply tools/code_format/check_format.py:166]: (deleted suggestion after reading actual file)

File: tools/code_format/check_format.py [hunk 2]
```diff
@@ -159,6 +159,16 @@ def evaluate_lines(self, path, line_xform, write=True):
...
+                elif line.find("#pragma once") == -1:
```
Comment (phlax) [inline tools/code_format/check_format.py:167]:
```suggestion
                elif "#pragma once" in line:
```

(i dislike this file soo much 8/)

File: tools/code_format/check_format.py [hunk 3]
```diff
@@ -159,6 +159,16 @@ def evaluate_lines(self, path, line_xform, write=True):
         format_flag = True
         output_lines = []
         for line_number, line in enumerate(self.read_lines(path)):
+            if line_number == 0 and path.endswith(".h"):
+                if line == "":
+                    error_message = "%s:%d: %s" % (
+                        path, line_number + 1, "the first line is empty for header files")
```
Comment (phlax) [inline tools/code_format/check_format.py:165]:
```suggestion
                    error_message =  f"{path}:{line_number + 1}: the first line is empty for header files"`

File: tools/code_format/check_format.py [hunk 4]
```diff
@@ -159,6 +159,16 @@ def evaluate_lines(self, path, line_xform, write=True):
...
+                    continue
+                elif line.find("#pragma once") == -1:
+                    error_message = "%s:%d: %s" % (
+                        path, line_number + 1, "#pragma once missed for header files")
```
Comment (phlax) [inline tools/code_format/check_format.py:169]: same as above - prefer `f` strings

File: tools/code_format/check_format.py (hunk 2 above)
Comment (phlax) [reply tools/code_format/check_format.py:167]: even better `not line.startswith("#pragma once")

**Review by phlax — APPROVED** [review]
lgtm, thanks @botengyao

File: tools/code_format/check_format.py (hunk 2 above)
Comment (botengyao) [reply tools/code_format/check_format.py]: Thanks! Let's write Pythonic code.

**Pull request merged**